### ISSUE TITLE
Suspicious behavior in generateTiddlerFilename 

### ISSUE THREAD

**Issue by dullroar** [Jermolene/TiddlyWiki5#1308]
Note the following:

``` javascript
// First remove any of the characters that are illegal in Windows filenames
var baseFilename = title.replace(/<|>|\:|\"|\/|\\|\||\?|\*|\^|\s/g,"_");
// Truncate the filename if it is too long
if(baseFilename.length > 200) {
    baseFilename = baseFilename.substr(0,200);
}
// Start with the base filename plus the extension
var filename = transliterate(baseFilename) + extension,
count = 1;
// Add a discriminator if we're clashing with an existing filename
while(existingFilenames.indexOf(filename) !== -1) {
filename = baseFilename + " " + (count++) + extension;
}
return filename;
```

It appears that if the "discriminator" code (which I am working on fixing for the other issue) executes, then the `transliterate` results are lost on the re-generated `filename`. It would seem more likely that `transliterate` should be run once and the results of that saved as `baseFilename`. Concurrence? If so I will fix it while I am in the code anyway.

**Comment by Jermolene** [comment]
Hi @dullroar good catch; it's possible that a problem was introduced with #1280.

**Comment by dullroar** [comment]
My proposal is to change:
```javascript
var baseFilename = title.replace(/<|>|\:|\"|\/|\\|\||\?|\*|\^|\s/g,"_");
```
...to:
```javascript
var baseFilename = transliterate(title.replace(/<|>|\:|\"|\/|\\|\||\?|\*|\^|\s/g,"_"));
```
If you agree I will make the change since I have that function open now and ***think*** I am about done with the fix for the Windows file name issue.

**Comment by Jermolene** [comment]
Thanks @dullroar that looks reasonable.

**Comment by pmario** [comment]
@dullroar ... has there been a PR for this issue?

**Comment by dullroar** [comment]
I believe it was fixed with PR #1312.
    
    
    On 05/17/2017 06:40 AM, Mario Pietsch
      wrote:
    
    
      @dullroar ... has there been a PR for
        this issue?
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Jermolene/TiddlyWiki5","title":"Jermolene/TiddlyWiki5","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Jermolene/TiddlyWiki5"}},"updates":{"snippets":[{"icon":"PERSON","message":"@pmario in #1308: @dullroar ... has there been a PR for this issue?"}],"action":{"name":"View Issue","url":"https://github.com/Jermolene/TiddlyWiki5/issues/1308#issuecomment-302064484"}}}
    
    
    -- 
Jim

“It's not about the nail!"

**Comment by pmario** [comment]
so you can close this issue?

**Comment by dullroar** [comment]
Fixed in PR #1312.
